Upgrade assistant - Phase 1

joshdover opened the pull request:
## Summary

Closes #26210

- Adds a "7.0 Upgrade Assistant" to the Elasticsearch section of the Management app.
- UI shows deprecation warnings for cluster and nodes
- Shows a button to open the reindexing steps in the Console

![image](https://user-images.githubusercontent.com/1813008/49598488-6016cf00-f944-11e8-8eb8-dc604c6139a9.png)
![image](https://user-images.githubusercontent.com/1813008/49598537-72910880-f944-11e8-8144-0a6f87c85b93.png)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~ (Being handled by @gchaps)
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [x] ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

joshdover committed:
Get basic scaffolding working

elasticmachine commented:
Pinging @elastic/kibana-operations

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/7716/)

joshdover committed:
Wire up cluster checkup data

joshdover committed:
Add types for elasticsearch plugin

joshdover committed:
Implement basics of checkup tab

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/8237/)

joshdover committed:
Update style of deprecations

joshdover committed:
Add copy and reload button to checkup

joshdover committed:
Add filtering by warning level

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/8262/)

joshdover committed:
Add deprecation logging tab

joshdover committed:
Copy updates and cleanup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/8350/)

joshdover committed:
Type cleanup

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Move deprecation logging to overview tab

joshdover committed:
Make filters togglable

joshdover committed:
Move sections into tabs and add support for grouping

joshdover committed:
Cleanup and add clearer labels

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/88/)

joshdover committed:
Use tables for message grouping

joshdover commented:
I've updated this to use tables when grouping by issue. This should help with customers who have a thousands of indices that have the same problem. With a table we get pagination and sorting that allows them easily see issues without blowing up the size of the page while also being able to quickly see index-specific details in the right column.

![image](https://user-images.githubusercontent.com/1813008/48586543-db610400-e8f5-11e8-8ae5-f0a9dc558db2.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/269/)

joshdover committed:
Cleanup and small fixes

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/587/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/589/)

joshdover committed:
Allow console to load relative URLs

joshdover committed:
Add reindex in console button to reindex tasks

joshdover committed:
Merge documentation UI and uiButtons

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/716/)

joshdover committed:
Fix tests

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/748/)

tylersmalley commented on x-pack/plugins/upgrade_checkup/server/lib/es_migration_apis.ts:
```diff
@@ -0,0 +1,136 @@
+  // ]);
+
+  // Fake data for now. TODO: Uncomment above and remove fake data
+  const migrationAssistance = _.cloneDeep(fakeAssistance) as AssistanceAPIResponse;
```
Need to import lodash.

joshdover replied:
Really confused on how this import got removed...apparently tslint sees lodash as a global?

joshdover committed:
Filter bar tweaks

joshdover committed:
Filter out index settings that can't be set

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/886/)

joshdover committed:
Fix types

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/890/)

joshdover commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/907/)

joshdover committed:
Add tests for deprecation_logging

joshdover committed:
Add tests for reindex templates

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Make KibanaConfig generic

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1261/)

joshdover commented:
retest

joshdover committed:
Simplify integration test

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1272/)

joshdover committed:
Finish backend unit tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1279/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1416/)

cchaos committed:
Fixed background color stretching

joshdover committed:
Fixup types

cchaos committed:
Pulling tabs out and re-ordering filter buttons

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1440/)

cchaos committed:
Making accordions more item-list like

cchaos committed:
Turned Healths into Badges

- Couldn’t do the conversion within the cell because it only passed color

cchaos committed:
Fix overflow issue

cchaos committed:
Optional filter and expand/collapse controls

cchaos committed:
Reorganizing

- Added placeholder for moving action button up into accordion header
- Removed repetitive message name outputs
- Slightly better listing of each message when sorting by index
- Only showing number of severity when sorting by index
  - Still need to allow showing all severity levels
- Added indice count when sorting by issue

cchaos committed:
Putting `Deprecation logging` in a `EuiDescribedFormGroup`

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Fix uiButton updating for reindex items

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1589/)

joshdover committed:
Update blacklisted settings

cchaos committed:
Added some stats, empty prompts, and all clear prompt

cchaos committed:
Added docs link

cchaos committed:
Cleaned up sass files

cchaos committed:
Revert changes to fake_deprecations

joshdover committed:
wip

joshdover committed:
Merge pull request #25 from cchaos/joshdover-upgrade-checkup

Upgrade Checker - Design PR

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2103/)

joshdover committed:
Move data fetching and tab control

joshdover committed:
Wire up overview summary

joshdover committed:
Cleanup docs/uiButtons + move actions to index table

joshdover committed:
Add expand/collapse all functionality

joshdover committed:
Wire up search box

joshdover committed:
Wire up severity indicators

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2180/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Fix types

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2335/)

joshdover committed:
Round out functional tests

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2395/)

joshdover committed:
Fix fake data

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Remove info deprecation level

joshdover committed:
Fix extra space on cluster tab control bar

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2513/)

joshdover committed:
Cleanup code and localize majority of UI controls

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2571/)

joshdover committed:
Change overview tab to steps layout

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2652/)

cchaos commented:
There still seems to be a missing piece about what happens if all the checks are good? Where do they go next.

@gjones was mentioning that Kibana should be able to recognize if the instance is on prem or in cloud, and if it's in cloud we should direct them to the `deployment overview page`.

@joshdover Can you see what you would need from cloud to be able to create a link to point the user to this page? And if on prem, where should we point them to to proceed with the upgrade?

joshdover committed:
Update copy

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2701/)

joshdover committed:
Localize overview tab

joshdover committed:
Complete localization of checkup tabs

joshdover committed:
Make ES version dynamic based on branch

joshdover committed:
Add pagination to checkup tabs

joshdover committed:
Rename checkup -> assistant

joshdover committed:
Cleanup filter and group by bars

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2841/)

joshdover committed:
WIP UI unit tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2851/)

joshdover committed:
Copy tweaks

joshdover committed:
Fix i18n formatting issues

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2941/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2968/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Update tests for copy

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3033/)

joshdover committed:
Add tests for remaining UI

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3080/)

cchaos commented on x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/index_table.tsx:
```diff
@@ -0,0 +1,160 @@
+        sorting={sorting}
+        pagination={pagination}
+        onChange={this.onTableChange}
+      />
```
Add `hasActions={false}` so that the buttons aren't desaturated. They currently look disabled.

 
<img width="983" alt="screen shot 2018-12-11 at 09 57 59 am" src="https://user-images.githubusercontent.com/549577/49810877-1e18cf00-fd30-11e8-820b-925e37edb281.png">

cchaos commented on x-pack/plugins/upgrade_assistant/public/components/tabs/overview/steps.tsx:
```diff
@@ -0,0 +1,196 @@
+        {
+          title: intl.formatMessage({
+            id: 'xpack.upgradeAssistant.overviewTab.steps.clusterStep.stepTitle',
+            defaultMessage: 'Check for issues with your cluster',
```
Can we make this messaging more succinct? Something like:

<img width="493" alt="screen shot 2018-12-11 at 10 30 19 am" src="https://user-images.githubusercontent.com/549577/49811004-55877b80-fd30-11e8-8a34-8c44f868fe52.png">

^^ @gchaps ?

cchaos reviewed (commented):
I found an issue with the combination of pagination and the filter bar where if you're on a later page (like the last one) then you type to filter, it renders nothing and you have to select an earlier page. When you update with the filter can you just always set the page number to 1?

<img src="https://d.pr/free/i/cmABdq+" />

----

Has there been any thoughts or progress on the "All clear" message that will direct them to perform the upgrade?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3190/)

joshdover commented:
@cchaos Nice catch on the pagination/filter issue, I'll get that fixed up today.

Working on the copy with @gchaps right now for adding a 4th step that would instruct the user to proceed with the upgrade.

joshdover committed:
Fix pagination w/ filter changes + table button color

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3204/)

joshdover commented on src/core_plugins/elasticsearch/index.d.ts:
```diff
@@ -0,0 +1,108 @@
+ * under the License.
+ */
+
+export class Cluster {
```
This will be replaced by work in #26045

joshdover committed:
Small cleanup

joshdover committed:
Add reindex button to old index deprecations

joshdover committed:
Add shrunken indices setting to copy settings blacklist for #18469

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3281/)

joshdover committed:
Add next steps to overview tab + update copy

joshdover committed:
Remove usage of migration assistance API

joshdover committed:
Use all/critical toggle for filter buttons

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Cloud upgrade copy

joshdover committed:
Translate reindex button

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3479/)

joshdover committed:
Remove hacked EUI type

joshdover committed:
Show incomplete banner on all tabs

joshdover committed:
Update copy for waiting for next version

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3715/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3734/)

tylersmalley commented on x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts:
```diff
@@ -0,0 +1,114 @@
+                {
+                  defaultMessage: 'Reindex in Console',
+                  description:
+                    '"Console" should be the same name used to label the app under Dev Tools -> Console',
```
Are the actions something which we are introducing in Kibana, or does ES also expose the deprecations that way?

joshdover replied:
Only in Kibana right now.

tylersmalley replied:
Thoughts about moving this check to the UI? - my thinking is that it would drastically cut down on the payload size. Additionally, including it in the API means we have to be cognisent of API breaking changes going forward.

tylersmalley replied:
note: Actions were removed from the API

tylersmalley commented on x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts:
```diff
@@ -0,0 +1,114 @@
+      .map(d => ({ ...d, index: indexName } as EnrichedDeprecationInfo))
+      .map(d => {
+        // Add reindexing action if it's the old index deprecation warning.
+        if (d.message === `Index created before ${CURRENT_MAJOR_VERSION}.0`) {
```
Is this still the message from ES for when a reindex is required?

As an aside, I am not seeing indices required for reindex being returned by the ES deprecations API. I don't see that being tracked in https://github.com/elastic/elasticsearch/issues/36024. Do we know what the status of that is?

joshdover replied:
Yes this is still the message. I was able to get indices created in 5.6 returned from ES as old from a 6.6 snapshot. Haven't tried against a 7.0.0 snapshot actually.

tylersmalley replied:
👍 I tested an index created in 6.6 against 7.0 - will try 5.6-> 6.6.

tylersmalley reviewed (dismissed):
Stopping point for the day - so far looks good, just a couple things.

sqren commented on x-pack/plugins/apm/server/lib/helpers/setup_request.ts:
```diff
@@ -6,28 +6,13 @@
-    config: () => KibanaConfig;
-  }
-}
-
```
Nice, thank you!

sorenlouv reviewed (approved):
Changes to APM look good 👍

joshdover commented on x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts:
```diff
@@ -0,0 +1,114 @@
+  callWithRequest: CallClusterWithRequest,
+  req: Request,
+  basePath: string,
+  useFakeData = false // TODO: remove
```
Yeah I'll go ahead and drop this.

joshdover commented on x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/constants.tsx:
```diff
@@ -0,0 +1,24 @@
+
+export const LEVEL_MAP = {
+  none: 0,
+  info: 1,
```
True, we can remove none and info at this point.

joshdover commented on test/functional/page_objects/settings_page.js:
```diff
@@ -53,6 +53,10 @@ export function SettingsPageProvider({ getService, getPageObjects }) {
     }
 
+    async clickElasticsearchUpgradeAssistant() {
+      await testSubjects.click('upgrade_assistant');
```
Good thought, I'll move this.

joshdover commented on x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/index_table.tsx:
```diff
@@ -0,0 +1,161 @@
+import { EuiBasicTable, EuiButton } from '@elastic/eui';
+import { injectI18n } from '@kbn/i18n/react';
+
+const PAGE_SIZES = [10, 25, 50, 100, 250, 500, 1000];
```
Maybe I made this a bit too smart. I made it so it actually only shows the options that make sense (eg. if there's 35 options, only show 10, 25, and 50, don't show the bigger ones). We could probably remove this smartness, but I thought it was easy enough to implement so I did.

joshdover committed:
Review comments

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3895/)

joshdover committed:
Update deprecation level type

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3921/)

joshdover committed:
Update checkup tab snapshots

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3949/)

joshdover committed:
Remove dependencies on types from #25168

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4135/)

joshdover committed:
Merge remote-tracking branch 'upstream/master' into upgrade-checkup

joshdover committed:
Use types from new global type defs

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4187/)

tylersmalley commented on x-pack/plugins/upgrade_assistant/public/components/tabs/checkup/deprecations/list.test.tsx:
```diff
@@ -0,0 +1,128 @@
+
+    test('shows simple messages when index field is not present', () => {
+      expect(shallow(<DeprecationList {...defaultProps} />)).toMatchInlineSnapshot(`
+<div>
```
`dedent` is nice to keep this inline - not necessary to change, just an FYI.

tylersmalley commented:
I am not seeing the open in console button show up for indices which need re-indexed.

joshdover commented:
> I am not seeing the open in console button show up for indices which need re-indexed.

Ah yeah it won't work when you're running off master since it's looking for the 7.0 -> 8.0 upgrade message. You can override this by changin this file: `x-pack/plugins/upgrade_assistant/common/version.ts`

joshdover committed:
Remove 'Reindex in Console' button

tylersmalley reviewed (approved):
LGTM

tylersmalley commented:
As discussed, we removed console reindexing templates until we ensure it's the correct guidance.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4468/)

tylersmalley commented on x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.test.ts:
```diff
@@ -0,0 +1,41 @@
+import { getUpgradeAssistantStatus } from './es_migration_apis';
+
+import { DeprecationAPIResponse } from 'src/legacy/core_plugins/elasticsearch';
+import { CURRENT_MAJOR_VERSION } from '../../common/version';
```
Looks like this can be removed.

joshdover committed:
Remove unused variable

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4479/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4508/)

joshdover commented:
6.x: https://github.com/elastic/kibana/commit/6721466e9a001a32953cb0fcded11ebb248c799c
